public/app/features/dashboard-scene/scene/DashboardScene.test.tsx

```diff
@@ -75,28 +75,37 @@ describe('DashboardScene', () => {
         }
       );
 
-      // TODO: Make the dashboard to restore the defaults on discard
-      it.skip('A change to refresh picker interval settings should set isDirty true', () => {
+      it('A change to refresh picker interval settings should set isDirty true', () => {
         const refreshPicker = dashboardSceneGraph.getRefreshPicker(scene)!;
-        const prevState = refreshPicker.state.intervals;
         refreshPicker.setState({ intervals: ['10s'] });
 
         expect(scene.state.isDirty).toBe(true);
 
-        scene.onDiscard();
-        expect(refreshPicker.state.intervals).toEqual(prevState);
+        // TODO: Make the dashboard to restore the defaults on discard
+        // scene.onDiscard();
```

Comment:
Isn't this working already (to an extent)? We are capturing the initial state of a scene before navigating to edit mode and discarding on edit mode close (https://github.com/grafana/grafana/blob/support-hide-timerange/public/app/features/dashboard-scene/scene/DashboardScene.tsx#L174)

Reply:
I mean - it at least work in the UI:)

Reply:
It doesn't work for nested objects. It works for the dashboard scene state but not for the time picker state, the dashboard control state, etc; since their state is not restored.